[benson1029] iP

benson1029 opened the pull request:
### Echon

> "Punctuality is the politeness of kings." - Louis XVIII of France [(source)](https://www.brainyquote.com/topics/punctuality-quotes)

Looking for a tool to remind you to do things on time? **Echon is what you need**! It's

- text-based
- ~~easy~~ VERY EASY to use

All you need to do is,

1. download it from the release page.
2. double-click it.
3. add your tasks.
4. let it manage tasks for you. 😄 

Features:
- [X] Adding tasks
- [X] Keeping track of deadline of tasks
- [ ] Export to other formats (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) {
    ui = new UI();
    try {
        storage = new Storage(FILE_PATH);
        taskList = storage.load();
    } catch (DukeException e) {
        ui.printMessage("Error loading task list: " + e.getMessage());
        return;
    }
    commandCreator = new CommandCreator(taskList);
    ui.printWelcomeMessage();
    ui.processCommands(commandCreator, storage);
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

benson1029 committed:
Rename chatbot to Echon

benson1029 committed:
Add support for echo and bye commands

benson1029 committed:
Add task list related commands

benson1029 committed:
Add mark as done command

benson1029 committed:
Add support for three task types

benson1029 committed:
Reorganize code into files

benson1029 committed:
Edit automated text UI tests

benson1029 committed:
Handle errors in commands

benson1029 committed:
Add delete task command

benson1029 committed:
Save tasks in hard disk after each command

benson1029 committed:
Merge branch 'branch-Level-7'

benson1029 committed:
Change date and time format

benson1029 committed:
Merge branch 'branch-Level-8'

benson1029 committed:
Organize class into package duke

benson1029 committed:
Merge branch 'add-gradle-support'

benson1029 committed:
Add gradle support

benson1029 committed:
Add JUnit tests for Deadline and TaskList

benson1029 committed:
Add JavaDoc for public classes

benson1029 committed:
Add checkstyle

benson1029 committed:
Add command to find tasks by keyword

benson1029 committed:
Merge branch 'branch-A-JavaDoc'

benson1029 committed:
Merge branch 'branch-A-CodingStandard'

benson1029 committed:
Merge branch 'branch-Level-9'

benson1029 committed:
Rename package name from duke to echon

benson1029 committed:
Add basic user interface for Echon

benson1029 committed:
Merge branch 'branch-Level-10'

chenyixin0 commented on src/main/java/echon/Command.java:
```diff
@@ -0,0 +1,220 @@
+}
+
+class AddDeadlineCommand extends AddTaskCommand {
+    private String byDate;
```
Maybe consider renaming this field to dueDate for greater clarity?

benson1029 replied:
I was originally trying to follow the command format. But I agree that this variable name sounds unclear. I've renamed it to dueDate. Thanks for bringing that out!

chenyixin0 reviewed (commented):
Overall, very well written with good adherence to the Java coding standard. Good incorporation of images in the chatbot as well!

benson1029 committed:
Add assertions to the code

Several assumptions are made when processing commands, tasks and
storage. These assumptions are not checked and can lead to
unexpected behavior.

Assertions are added in these places to check the assumptions. This
will help to catch bugs early and make the code more robust.

benson1029 committed:
Improve code quality

Serveral classes have been refactored to improve code quality. This
includes renaming of the deadline byDate to dueDate, and the
separation of task finding and message construction into its own
methods. Some JavaDoc documentation errors have also been fixed.

benson1029 committed:
Merge pull request #1 from benson1029/branch-A-Assertions

Add assertions to the code

benson1029 committed:
Merge branch 'master' into branch-A-CodeQuality

benson1029 committed:
Merge pull request #2 from benson1029/branch-A-CodeQuality

Improve code quality

benson1029 committed:
Set up CI using GitHub actions

benson1029 committed:
Support tentative scheduling for events

Events could only be scheduled to a specific date and time.

This commit adds the ability to schedule events tentatively, which means
that the event is not yet scheduled to a specific date and time.
The user is able to supply multiple tentative start and end times for
the event, and once a confirm command is issued, the tentative event
will become an event object with only one start and end time.

As a tentative event involves a data structure that is different from
the regular event, a new class TentativeEvent is created to handle
tentative events. An Event object is created from a TentativeEvent
object when the user confirms the tentative event.

benson1029 committed:
Merge branch 'branch-B-TentativeScheduling'

benson1029 committed:
Improve code quality

benson1029 committed:
Handle non-integral or out of bound index in commands

benson1029 committed:
Add unit tests for tentative event

benson1029 committed:
Add product screenshot

benson1029 committed:
Add user guide for Echon

gowthaman-01 commented on src/main/java/echon/Command.java:
```diff
@@ -0,0 +1,293 @@
+/**
+ * Represents a command that can be executed by the user.
+ */
+public abstract class Command {
```
Avoid declaring multiple Java classes in a single file.

gowthaman-01 commented on src/main/java/echon/Task.java:
```diff
@@ -0,0 +1,53 @@
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
```
Try to reduce the use of this. Use of this is discouraged, potentially to avoid unnecessary noise in the code. Try to avoid using this except when it is shadowed by a method or constructor parameter.

gowthaman-01 commented on src/main/java/echon/TaskList.java:
```diff
@@ -0,0 +1,107 @@
+    /**
+     * Returns the number of tasks in the task list.
+     *
+     * @return The number of tasks in the task list.
```
The `@return` doesn't add any meaning, so you can remove it.

gowthaman-01 commented on src/main/java/echon/TaskBuilder.java:
```diff
@@ -0,0 +1,57 @@
+        String description = lineSplit[2];
+        Task task;
+        switch (type) {
+        case "T":
```
Try using an enum instead of matching strings

benson1029 committed:
Improve GUI of dialog box

benson1029 committed:
Update product screenshot

benson1029 committed:
Add AI use record
